jwst/associations/lib/tests/test_update_path.py

```diff
@@ -0,0 +1,60 @@
+from copy import deepcopy
+from jwst.associations.lib.update_path import update_key_value, replace_path
+
+_test_obj = {
+    "a": "change",
+    "b": "nochange",
+    "c": {"a": "change", "b": "nochange"},
+    "d": [
+        {"a": "change", "b": "nochange"},
+        {"a": "change", "b": "nochange"},
+    ],
+}
+
+
+def _gen_dict_extract(key, var):
+    if hasattr(var, "items"):
+        for k, v in var.items():
+            if k == key:
+                yield v
+            if isinstance(v, dict):
+                for result in _gen_dict_extract(key, v):
+                    yield result
```

Comment:
These could be changed to `yield from`, although it's fine to leave as-is because we're mostly not style-checking test modules